mantle/kola/harness.go

```diff
@@ -642,41 +659,23 @@ func runProvidedTests(tests map[string]*register.Test, patterns []string, multip
 	firstRunErr := suite.Run()
 	firstRunErr = handleSuiteErrors(outputDir, firstRunErr)
 
-	if len(failedTests) > 0 && rerun {
+	var rerunPatterns []string
+	for _, h := range failedTests.getArr() {
+		if h.Failed() && h.Name() != "non-exclusive-tests" {
+			name := strings.TrimPrefix(h.Name(), "non-exclusive-tests/")
+			rerunPatterns = append(rerunPatterns, name)
+		}
+	}
+	if len(rerunPatterns) > 0 && rerun {
 		newOutputDir := filepath.Join(outputDir, "rerun")
-		err = rerunFailedTests(flight, newOutputDir, pltfrm, versionStr)
-		handleSuiteErrors(newOutputDir, err)
+		fmt.Printf("\n\n======== Re-running failed tests (flake detection) ========\n\n")
+		runProvidedTests(testsBank, rerunPatterns, multiply, false, pltfrm, newOutputDir, propagateTestErrors)
 	}
 
 	// If the intial run failed and the rerun passed, we still return an error
 	return firstRunErr
 }
 
-func rerunFailedTests(flight platform.Flight, outputDir string, pltfrm string, versionStr string) error {
```

Comment:
this function is commented out in the first commit and deleted in the second commit. Let's delete it in the first commit unless there was a reason to do it like this.